### PR TITLE
Share code in fields of queries

### LINKED ISSUE #983: Reduce duplication in Commands/Query.hs
While working on `query proposals`, I noticed that many many fields are duplicated in a number of records in https://github.com/IntersectMBO/cardano-cli/blob/82160245479ff2e0c89d1d75f16fb39902aef466/cardano-cli/src/Cardano/CLI/EraBased/Commands/Query.hs#L85

This issue keeps track of sharing all these fields in a nested record.

### PR DESCRIPTION
**Pull request by smelc**
# Changelog

```yaml
- description: |
    Share code in fields of queries
# uncomment types applicable to the change:
  type:
  # - feature        # introduces a new feature
  # - breaking       # the API has changed in a breaking way
  # - compatible     # the API has changed but is non-breaking
  # - optimisation   # measurable performance improvements
  - refactoring    # QoL changes
  # - bugfix         # fixes a defect
  # - test           # fixes/modifies tests
  # - maintenance    # not directly related to the code
  # - release        # related to a new release preparation
  # - documentation  # change in code docs, haddocks...
```

# Context

Fixes https://github.com/IntersectMBO/cardano-cli/issues/983

# How to trust this PR

* This is one of the PRs where, if it typechecks, it works.
* -292+225

# Checklist

- [X] Commit sequence broadly makes sense and commits have useful messages
- [X] Self-reviewed the diff

### CODE REVIEW

File: cardano-cli/test/cardano-cli-golden/files/golden/help/allegra_query_kes-period-info.cli [hunk 1]
```diff
@@ -24,6 +24,6 @@ Available options:
                            CARDANO_NODE_NETWORK_ID environment variable
   --testnet-magic NATURAL  Specify a testnet magic id. This overrides the
                            CARDANO_NODE_NETWORK_ID environment variable
+  --out-file FILEPATH      The output file.
   --op-cert-file FILEPATH  Filepath of the node's operational certificate.
-  --out-file FILEPATH      Optional output file. Default is to write to stdout.
```
Comment (Jimbo4350) [inline cardano-cli/test/cardano-cli-golden/files/golden/help/allegra_query_kes-period-info.cli]: Why the change? It was more informative before.

File: cardano-cli/test/cardano-cli-golden/files/golden/help/allegra_query_stake-address-info.cli [hunk 2]
```diff
@@ -1,11 +1,11 @@
 Usage: cardano-cli allegra query stake-address-info --socket-path SOCKET_PATH
                                                       [--cardano-mode
                                                         [--epoch-slots SLOTS]]
-                                                      --address ADDRESS
                                                       ( --mainnet
                                                       | --testnet-magic NATURAL
                                                       )
                                                       [--out-file FILEPATH]
+                                                      --address ADDRESS
```
Comment (Jimbo4350) [inline cardano-cli/test/cardano-cli-golden/files/golden/help/allegra_query_stake-address-info.cli:7]: This is the downside of the refactor. It affects the parsers and I would argue having `--out-file` as the last argument is more natural. More and more I wish for monadic parsers.

**Review by Jimbo4350 — APPROVED** [review]
We should exclude the output file from the refactor. It's a UI wart IMO but the rest of the refactor is :+1:. I'm sure @CarlosLopezDeLara  would agree here. I also think leaving the original `out-file` help is more informative.

File: cardano-cli/test/cardano-cli-golden/files/golden/help/allegra_query_kes-period-info.cli (hunk 1 above)
Comment (smelc) [reply cardano-cli/test/cardano-cli-golden/files/golden/help/allegra_query_kes-period-info.cli]: Reverted this to use the better message :+1:

File: cardano-cli/test/cardano-cli-golden/files/golden/help/allegra_query_stake-address-info.cli (hunk 2 above)
Comment (smelc) [reply cardano-cli/test/cardano-cli-golden/files/golden/help/allegra_query_stake-address-info.cli:7]: Yes I agree too and was wondering after the initial golden files changes myself.

I pulled `mOutFile` out of the `QueryCommons` type and made sure that `--out-file` is the last argument in every query :+1:

File: cardano-cli/test/cardano-cli-golden/files/golden/help/allegra_query_kes-period-info.cli (hunk 1 above)
Comment (smelc) [reply cardano-cli/test/cardano-cli-golden/files/golden/help/allegra_query_kes-period-info.cli]: (the difference was between using `optional pOutputFile` and `pMaybeOutputFile`. The latter has the better message and I made sure to use it everywhere now)
